app/code/core/Mage/Reports/Model/Resource/Report/Collection/Abstract.php

```diff
@@ -268,4 +268,48 @@ public function load($printQuery = false, $logQuery = false)
         }
         return parent::load($printQuery, $logQuery);
     }
+
+
+    /**
+     * Get SQL for get record count
+     *
+     * @return Varien_Db_Select
+     * @see Mage_Reports_Model_Resource_Report_Product_Viewed_Collection
+     * @see Mage_Sales_Model_Resource_Report_Bestsellers_Collection
+     */
+    public function getSelectCountSql()
+    {
+        $this->_renderFilters();
+        $select = clone $this->getSelect();
+        $select->reset(Zend_Db_Select::ORDER);
+        return $this->getConnection()->select()->from($select, 'COUNT(*)');
+    }
+
+    /**
+     * Set ids for store restrictions
+     *
+     * @param  array $storeIds
+     * @return $this
+     * @see Mage_Reports_Model_Resource_Report_Product_Viewed_Collection
+     * @see Mage_Sales_Model_Resource_Report_Bestsellers_Collection
+     */
+    public function addStoreRestrictions($storeIds)
+    {
+        if (!is_array($storeIds)) {
+            $storeIds = [$storeIds];
+        }
+        $currentStoreIds = $this->_storesIds;
```

Comment:
this should be in a separate PR for easier review

Reply:
For some reason i dont see that comment/review in history?

It moved 1:1 duplicated methods to its parent abtract class.

Reply:
Its one commit https://github.com/OpenMage/magento-lts/pull/4484/commits/cb3fbbac519c1ba186db138556cf9367cd28bf71